super-csv/src/main/java/org/supercsv/io/Tokenizer.java

```diff
@@ -228,32 +231,59 @@ else if( c == quoteChar ) {
 					potentialSpaces = 0;
 					currentColumn.append(c);
 				}
-				
+
 			} else {
-				
+
 				/*
 				 * QUOTE_MODE (within quotes).
 				 */
-				
+
 				if( c == quoteChar ) {
```

Comment:
Good catch @jamesbassett -- I added a test case for that situation and decided to throw an exception when a field is quoted in both doubled-up quote style as well as escaped quote style.

I also switched the order of the blocks to handle the `c == quoteEscapeChar` block first, though that gets a little tricky when `quoteEscapeChar == quoteChar` as is the default.  In that case a lone `quoteChar` needs to be interpreted as a lone quote rather than a lone escape char, otherwise many tests break.